### DISCUSSION TITLE
Package Usage

### DISCUSSION THREAD

**Discussion by EyalAbadi** [coady/graphique#9] [General]
Hello,

I'm trying to understand how to use this package. Do I need to have a regular GraphQL running app and then use this package there or is this standalone?

If so, how do I integrate it?

Thanks!

**Comment by coady** [comment]
> I'm trying to understand how to use this package. Do I need to have a regular GraphQL running app and then use this package there or is this standalone?

It's standalone. `pip install graphique[server]` includes everything needed to run it as a service: `env PARQUET_PATH=... uvicorn graphique.service:app`

> If so, how do I integrate it?

It could also be used as a package, but I'd have to know more about the use case.

**Comment by EyalAbadi** [comment]
Hi Coady,

I'm trying to expose a parquet file stored on disk and expose it through a GraphQL type of API. Allowing the client to specify their required schema and return/filter the files. I'm trying to stand up a basic instance of it, but getting confused as I'm used to using something like apollo with its specific graphql terms and structure.

**Comment by coady** [comment]
For apollo federation, the schema can be output for composition with: `python3 -m graphique.schema <parquet-path>`.

**Comment by coady** [comment]
b677fb9 adds a `FEDERATED` configuration option.

**Comment by coady** [comment]
[v0.7](https://github.com/coady/graphique/releases/tag/v0.7) releases the federated option.

But I am also interested in this project being usable as extensible package: multiple tables, multiple indexes, and custom fields. Converting to a disussion for follow-up.

**Comment by coady** [comment]
[v1.0](https://github.com/coady/graphique/releases/tag/v1.0) is more extensible. A custom `GraphQL` app can be created, as in `tests/federated.py`. Or it can be used as a package, extending the `Dataset` interface.
